e2e/runTests.js

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable no-await-in-loop */
+
+const cypress = require('cypress');
+const fse = require('fs-extra');
+const {merge} = require('mochawesome-merge');
+const generator = require('mochawesome-report-generator');
+
+async function runTests() {
+    await fse.remove('mochawesome-report'); // remove the report folder
+    await fse.remove('results');
+    await fse.remove('screenshots');
+
+    const testDirs = fse.readdirSync('cypress/integration/');
+    let totalFailed = 0;
+
+    for (const dir of testDirs) {
+        const {failed} = await cypress.run({
+            spec: `./cypress/integration/${dir}/**/*`,
+            config: {
+                screenshotsFolder: 'screenshots',
+                trashAssetsBeforeRuns: false,
+            },
+            reporter: 'cypress-multi-reporters',
+            reporterOptions:
+                {
+                    reporterEnabled: 'mocha-junit-reporters, mochawesome',
+                    mochaJunitReportersReporterOptions: {
+                        mochaFile: 'results/junit/test_results[hash].xml',
+                        toConsole: false,
+                    },
+                    mochawesomeReporterOptions: {
+                        reportDir: 'mochawesome-report',
+                        reportFilename: `mochawesome-${dir}`,
+                        quiet: true,
+                        overwrite: false,
+                        html: false,
+                        json: true,
+                    },
+                },
+        });
+
+        totalFailed += failed;
+    }
+
+    const jsonReport = await merge(); // generate JSON report
+
+    await generator.create(jsonReport);
+
+    await fse.copy('screenshots', 'mochawesome-report/screenshots');
```

Comment:
0/5, could we expose a single top-level folder for all those auto-generated files?  Maybe all in `/e2e/reports`; folder structure could be:
```
|-- e2e
    |-- report
        |-- mochawesome-report
        |-- results
        |-- screenshots
```

One thing that it solves is to easily ignore-pattern for ESLint rules (`--ignore-pattern e2e/report`), as it's complaining too much when you ran ESLint with files under `e2e/mochawesome-report`, specifically the `app.js` on it.

Reply:
Will put everything in results/ directory. Will update the pipeline PR as well after I verify some of the other kinks.